services/agents/registry.go

```diff
@@ -456,6 +459,65 @@ func (r *Registry) SendSetStateRequest(ctx context.Context, pmmAgentID string) {
 	l.Infof("SetState response: %+v.", resp)
 }
 
+// CheckConnectionToService sends request to pmm-agent to check connection to service.
+func (r *Registry) CheckConnectionToService(ctx context.Context, service *models.Service, agent *models.Agent) error {
+	// TODO: extract to a separate struct to keep Single Responsibility principles.
+	l := logger.Get(ctx)
+	start := time.Now()
+	defer func() {
+		if dur := time.Since(start); dur > time.Second {
+			l.Warnf("CheckConnectionToService took %s.", dur)
+		}
+	}()
+
+	pmmAgentID := pointer.GetString(agent.PMMAgentID)
+	pmmAgent, err := r.get(pmmAgentID)
+	if err != nil {
+		l.Infof("CheckConnectionToService: pmm-agent with ID %q is not currently connected.", pmmAgentID)
+		return err
+	}
+	var request *agentpb.CheckConnectionRequest
+
+	switch service.ServiceType {
+	case models.MySQLServiceType:
+		request = &agentpb.CheckConnectionRequest{
+			Type: inventorypb.ServiceType_MYSQL_SERVICE,
+			Dsn:  mysqlDSN(service, agent),
+		}
+	case models.PostgreSQLServiceType:
+		request = &agentpb.CheckConnectionRequest{
+			Type: inventorypb.ServiceType_POSTGRESQL_SERVICE,
+			Dsn:  postgresqlDSN(service, agent),
+		}
+	case models.MongoDBServiceType:
+		request = &agentpb.CheckConnectionRequest{
+			Type: inventorypb.ServiceType_MONGODB_SERVICE,
+			Dsn:  mongoDSN(service, agent),
+		}
+	default:
+		l.Panicf("unhandled Service type %s", service.ServiceType)
+	}
+
+	l.Infof("CheckConnectionRequest: %+v.", request)
+	resp := pmmAgent.channel.SendRequest(request)
+	l.Infof("CheckConnection response: %+v.", resp)
+	checkConnectionResponse := resp.(*agentpb.CheckConnectionResponse)
+	if !checkConnectionResponse.Success {
+		return nil
+	}
+	return status.Error(codes.FailedPrecondition, checkConnectionResponse.Error)
+}
+
+func (r *Registry) get(pmmAgentID string) (*agentInfo, error) {
```

Comment:
An interesting thing, I have something like this in Actions branch:

https://github.com/percona/pmm-managed/blob/bc909b2f8b472b8827f352662c0c19e3bd7f96da/services/agents/registry.go#L345-L354

Maybe we should merge those things somehow?

Reply:
@DexterHD let's merge it once some of us merge his PR. :)